src/contracts/core/DelegationManager.sol

```diff
@@ -513,6 +732,57 @@ contract DelegationManager is Initializable, OwnableUpgradeable, Pausable, Deleg
         return _operatorDetails[operator].stakerOptOutWindowBlocks;
     }
 
+    /**
+     * @notice Returns the number of actively-delegatable shares a staker has across all strategies
+     */
+    function getDelegatableShares(address staker) public view returns (IStrategy[] memory, uint256[] memory) {
+        // Get currently active shares and strategies for `staker`
+        uint256 podShares = eigenPodManager.podOwnerShares(staker);
+        (IStrategy[] memory strategyManagerStrats, uint256[] memory strategyManagerShares) 
+            = strategyManager.getDeposits(staker);
+
+        // Has shares in StrategyManager, but not in EigenPodManager
+        if (podShares == 0) {
+            return (strategyManagerStrats, strategyManagerShares);
+        }
+
+        IStrategy[] memory strategies;
+        uint256[] memory shares;
+
+        if (strategyManagerStrats.length == 0) {
+            // Has shares in EigenPodManager, but not in StrategyManager
+            strategies = new IStrategy[](1);
+            shares = new uint256[](1);
+            strategies[0] = beaconChainETHStrategy;
+            shares[0] = podShares;
+        } else {
+            // Has shares in both
+            
+            // 1. Allocate return arrays
+            strategies = new IStrategy[](strategyManagerStrats.length + 1);
+            shares = new uint256[](strategies.length);
+            
+            // 2. Place StrategyManager strats/shares in return arrays
+            for (uint256 i = 0; i < strategyManagerStrats.length; ) {
```

Comment:
nice to leave a note that this is inefficiency is only encountered by people who natively restake and liquid restake on the same address. we should discourage that in the docs. use different addresses.

Reply:
... surely this isn't too bad of a gas cost? It hasn't been intuitive to me that people should use different addresses - why do we want that?

Most people not tryna keep up with addresses like engineers.

Reply:
I think it's fine. I will say, by my estimation, worst case you end up with:
66*32 = 2112 memory expansion -- not sure on cost, depends on other memory already used
(66 * some small number) memory operations -- probably on the order of hundreds of gas